operator/cluster-operator.go

```diff
@@ -0,0 +1,135 @@
+package operator
+
+import (
+	crv1 "github.com/grtl/mysql-operator/pkg/apis/cr/v1"
+	v1beta2 "k8s.io/api/apps/v1beta2"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const MySQLPortNumber = 3306
+
+func AddCluster(cluster *crv1.MySQLCluster, kClientset kubernetes.Interface) {
+	createServiceForCluster(cluster, kClientset)
+	createStatefulSetForCluster(cluster, kClientset)
+}
+
+func createServiceForCluster(cluster *crv1.MySQLCluster, kClientset kubernetes.Interface) {
+	servicesInterface := kClientset.CoreV1().Services(cluster.ObjectMeta.Namespace)
+
+	newService := serviceForCluster(cluster)
+	_, err := servicesInterface.Create(&newService)
+
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		panic(err)
+	}
+}
+
+func createStatefulSetForCluster(cluster *crv1.MySQLCluster, kClientset kubernetes.Interface) {
+	statefulSetsInterface := kClientset.AppsV1beta2().StatefulSets(cluster.ObjectMeta.Namespace)
+
+	newStatefulSet := statefulSetForCluster(cluster)
+
+	_, err := statefulSetsInterface.Create(&newStatefulSet)
+
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		panic(err)
+	}
+}
+
+func serviceForCluster(cluster *crv1.MySQLCluster) v1.Service {
+	return v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cluster.Spec.Name,
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: "None",
+			Ports: []v1.ServicePort{
+				v1.ServicePort{
+					Port: MySQLPortNumber,
+				},
+			},
+			Selector: map[string]string{
+				"mysql-cluster": cluster.Spec.Name,
+			},
+		},
+	}
+}
+
+func statefulSetForCluster(cluster *crv1.MySQLCluster) v1beta2.StatefulSet {
```

Comment:
We might want to simplify this function, for example split the `container` and `pvc` definitions into separate functions